File: apps/web/src/types/vehicle.ts
export type Transmission = "MANUAL" | "AUTOMATIC" | "CVT";

export interface VehicleModifications {
  turbo: boolean | null;
  suspension: string | null;
  tune: string | null;
  injectors: string | null;
  fuel_type: string | null;
}

export interface Vehicle {
  id: string;
  brand: string;
  model: string;
  year: number;
  engine: string | null;
  vin: string | null;
  license_plate: string | null;
  color: string | null;
  mileage: number | null;
  transmission: Transmission | null;
  modifications: VehicleModifications | null;
  customer_id: string;
  customer_name: string;
  active: boolean;
  created_at: string;
}

export interface CreateVehicleRequest {
  brand: string;
  model: string;
  year: number;
  engine?: string;
  vin?: string;
  license_plate?: string;
  color?: string;
  mileage?: number;
  transmission?: Transmission;
  modifications?: {
    turbo?: boolean;
    suspension?: string;
    tune?: string;
    injectors?: string;
    fuel_type?: string;
  };
  customer_id: string;
}

export interface VehiclesPage {
  data: Vehicle[];
  meta: { total: number; page: number; limit: number };
}

File: apps/api/src/main/java/rd/tallerfacil/api/vehicle/domain/Vehicle.java
package rd.tallerfacil.api.vehicle.domain;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;
import rd.tallerfacil.api.customer.domain.Customer;
import rd.tallerfacil.api.shared.domain.TenantAwareEntity;

@Entity
@Table(name = "vehicles")
@Getter
@Setter
@NoArgsConstructor
public class Vehicle extends TenantAwareEntity {

    @Column(nullable = false, length = 100)
    private String brand;

    @Column(nullable = false, length = 100)
    private String model;

    @Column(nullable = false)
    private Integer year;

    @Column(length = 100)
    private String engine;

    @Column(length = 17)
    private String vin;

    @Column(name = "license_plate", length = 20)
    private String licensePlate;

    @Column(length = 50)
    private String color;

    private Integer mileage;

    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private Transmission transmission;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "jsonb")
    private VehicleModifications modifications;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    private Customer customer;

    @Column(nullable = false)
    private boolean active = true;
}
